include/SQLiteCpp/Statement.h

```diff
@@ -16,6 +16,25 @@
 #include <map>
 #include <climits> // For INT_MAX
 
+// some macros are taken from https://github.com/nemequ/hedley/blob/master/hedley.h , it was public domain that time
+#if defined(__GNUC__) || defined(__GNUG__) || defined(__clang__) ||\
+(defined(__INTEL_COMPILER) && __INTEL_COMPILER > 1600) ||\
+(defined(__ARMCC_VERSION) && __ARMCC_VERSION > 4010000) ||\
+(\
+    defined(__TI_COMPILER_VERSION__) && (\
+        __TI_COMPILER_VERSION__ > 8003000 ||\
+        (__TI_COMPILER_VERSION__ > 7003000 && defined(__TI_GNU_ATTRIBUTE_SUPPORT__))\
+    )\
+)
+    #if !defined(pure) and __has_attribute(const)
```

Comment:
oops, here is the error (though causing no harm in most of cases). Should be `!defined(SQLITECPP_PURE_FUNC)`